### PR TITLE
Add check for standard library/C++ standard version for the ranges testing, fix the main CMake file

### PR DESCRIPTION
**Pull request by evgeniusp**
[CMake]: 
* Remove unnecessary variables and macro from the main CMakeLists.txt: ONEDPL_USE_PARALLEL_POLICIES, ONEDPL_USE_RANGES_API
* Add check for the serial back-end

[Testing]
* Add check for C++ standard and standard library versions for the testing of ranges

### CODE REVIEW

**Review by AlexVeprev — APPROVED** [review]
CMake part looks good to me.

**Review by MikeDvorskiy — COMMENTED** [review]
I'm OK with macro _ONEDPL_USE_RANGES definition..
But CMake changes should be reviewed  a person who knows cmake quite well)..
Second question - @AlexVeprev, what about "skip" status in case of a configuration doesn't support the ranges ? Does the test launcher do such differentiation?

**Comment by AlexVeprev** [comment]
> what about "skip" status in case of a configuration doesn't support the ranges ? Does the test launcher do such differentiation?

Currently it is not differentiated. It's a separate topic, @dparanic works on it.

File: include/oneapi/dpl/ranges [hunk 1]
```diff
@@ -16,21 +16,29 @@
 #ifndef _ONEDPL_RANGES
 #define _ONEDPL_RANGES
 
+#if defined(_GLIBCXX_RELEASE)
```
Comment (capatober) [inline include/oneapi/dpl/ranges:19]: I see you don't include any standard library header. So, the `_GLIBCXX_RELEASE` macro shouldn't be defined. (See the example https://godbolt.org/z/vGebnv). I'd suggest the following:
```
// It should be done before any standard header inclusion
#if __has_include(<tbb/version.h>) 
#    ifndef PSTL_USE_PARALLEL_POLICIES
#        define PSTL_USE_PARALLEL_POLICIES (_GLIBCXX_RELEASE != 9)
#    endif
#    ifndef _GLIBCXX_USE_TBB_PAR_BACKEND
#        define _GLIBCXX_USE_TBB_PAR_BACKEND (_GLIBCXX_RELEASE > 10)
#    endif
#endif // __has_include(<tbb/version.h>)

#include<version>

#if defined(_GLIBCXX_RELEASE)
...
#endif
...
```

**Review by capatober — COMMENTED** [review]
Overall, patch looks good to me. Thanks.

File: include/oneapi/dpl/ranges (hunk 1 above)
Comment (evgeniusp) [reply include/oneapi/dpl/ranges:21]: Add TODO about the order of headers and inclusion of standard library headers version/ciso646.

**Review by capatober — APPROVED** [review]
Regarding headers ordering. We need to resolve it later

**Review by AlexVeprev — DISMISSED** [review]

**Review by capatober — APPROVED** [review]

**Review by capatober — APPROVED** [review]

**Pull request merged**